src/nxt_conf.c

```diff
@@ -783,12 +787,13 @@ nxt_conf_get_array_element(nxt_conf_value_t *value, uint32_t index)
 
 
 nxt_conf_value_t *
-nxt_conf_get_array_element_or_itself(nxt_conf_value_t *value, uint32_t index)
+nxt_conf_get_array_element_or_itself(const nxt_conf_value_t *value,
+    uint32_t index)
 {
     nxt_conf_array_t  *array;
 
     if (value->type != NXT_CONF_VALUE_ARRAY) {
-        return (index == 0) ? value : NULL;
+        return (index == 0) ? (nxt_conf_value_t *) value : NULL;
```

Comment:
The same as above, should the return type be with `const`?

Reply:
I'll check, but again this is likely to result in a much larger intrusive change.

Reply:
I can also drop this...

Reply:
Dropped.